CVLByExample/Curve/contracts/Curve.sol

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.13;
+import "./ERC20.sol";
+import "./ReentrancyGuard.sol";
+import "./CurveToken.sol";
+import "./CurveTokenExample.sol";
+
+contract MintableToken is ERC20 {
+
+    constructor (string memory name, string memory sym) ERC20(name, sym) {
+
+    }
+
+    function mint(address _user, uint _amount) public {
+        _mint(_user, _amount);
+    }
+}
+
+contract Curve is ReentrancyGuard{
+    //uint128 public constant  2 = 2;
+    uint256  constant A_PRECISION =100;
+    uint256  constant PRECISION = 10e18;
+    uint256  future_A_time;
+    uint256  future_A;
+    uint256  initial_A_time;
+    uint256  initial_A;
+    uint256[2]  admin_balances;
+    // storage var "address[2] coins" was seperated to be able to link
+    address  public coins_0;
+    address  public coins_1;
+    address  public lp_token;
+    address  owner;
+    // address  public underlying_token; // would be linked to ERC20 and equal to coins[1], but can't do directly due to CVL limitations
+    uint256  kill_deadline;
+    uint256  fee;
+    uint256  admin_fee;
+    uint256  future_fee;
+    uint256  future_owner;
+    uint256  future_admin_fee;
+    uint256  _DEMO_D;
+
+    mapping(uint256 => mapping(uint256 => mapping(uint256 => uint256))) _DEMO_D_MAPPING;
+
+
+
+    
+    // constructor(uint256 _future_A_time, uint256 _future_A, uint256 _initial_A_time, uint256 _initial_A, address owner ) {
+
+    constructor(address _lp_token, address _token_addr) payable {
+        lp_token = _lp_token;
+        coins_1 = _token_addr;
+    }
+
+
+    // function get_D(uint xp_0, uint xp_1)
+
+    function get_D(uint256[2] memory xp,uint256 amp) public view returns(uint256) {
+        uint256 S = 0;
+        uint256 Dprev = 0;
+        // Demo simplificaiton - unconstrained CONST
+        // get_D currently crashes the prover's pre-SMT analysis or something
+        // uint256 x; 
+        // return _DEMO_D; 
+        // return _DEMO_D_MAPPING[xp[0]][xp[1]][amp];
+        return xp[0] + xp[1];
+        // Simplification end
+
+        for (uint256 _x=0;_x<xp.length;_x++){
+            S +=xp[_x];
+        }
+        if(S == 0){
+            return 0;
+        }
+
+        uint256 D=S;
+        uint256 Ann = amp * 2;
+        for (uint _i=0;_i<255;++_i){
+            uint256 D_P = D;
+            for (uint256 _x=0;_x<xp.length;_x++){
+                D_P = D_P * D / (xp[_x] * 2 +1);
+            }
+            Dprev = D;
+            D = (Ann * S / A_PRECISION + D_P * 2) * D / ((Ann - A_PRECISION) * D / A_PRECISION + (2 + 1) * D_P);
+            if (D > Dprev){
+                if (D - Dprev <= 1){
+                    return D;
+                }
+            }
+            else{
+                if (Dprev - D <= 1){
+                    return D;
+                }
+            }
+        }
+        revert();
+    }
+
+    function _A() view internal returns(uint256) {
```

Comment:
I think we should add some documentation for this function.
Also, I would prefer a more meaningful name than `_A`.

Reply:
I don't understand the meaning of A, D and others. @DanieLion55 ?

Reply:
A, D are the names from the original Curve so better leave them. I added comments on what I know.